src/libraries/Common/src/Interop/Linux/OpenLdap/Interop.Ldap.cs

```diff
@@ -67,6 +68,26 @@ internal static partial class Ldap
     {
         static Ldap()
         {
+            // Register callback that tries to load other libraries when the default library "libldap-2.4.so.2" not found
+            AssemblyLoadContext.Default.ResolvingUnmanagedDll += (assembly, ldapName) =>
```

Comment:
Yep, you want to `GetLoadContext` for the current assembly. Otherwise things will still fail if the assembly is in a non-default ALC (and we'll be unnecessarily adding a handler to the defalut ALC).